metadata.yaml

```diff
@@ -6,6 +6,9 @@
 apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
 kind: Metadata
 releaseSeries:
+  - major: 1
+    minor: 2
+    contract: v1beta1
```

Comment:
@sbueringer  do we need to have the same change in test/e2e/data/shared/v1beta1/metadata.yaml too?

Reply:
Yes.

We need a few more changes in test data, but I think some of them would be better done after the v1.1 release.

But I think it's a good idea to keep the metadata.yaml in sync in this PR.

Reply:
I've just updated metadata in e2e, can you take a look?

Reply:
Looks perfect!

Reply:
yes, looks like CI won't be happy until 1.1 is released. Let's hold the PR until next week.

Reply:
+1 make a bulk of all the changes above

Reply:
I'll open an issue hopeful until EOW which captures/summarizes what we want to do. @alexander-demichev I'll mention you on it.